File: 3211-generate-binary-strings-without-adjacent-zeros/3211-generate-binary-strings-without-adjacent-zeros.cpp
class Solution {
public:
    void solve( int n , string str , vector<string> &result ){

        if( n == str.size() ){
            result.push_back(str);
            return ;
        }

        if( str.size() == 0 || str.back() == '1' ){
            solve( n , str + '0' , result );
        }

        solve( n , str + '1' , result );
        
    }
    vector<string> validStrings(int n) {
        
        vector<string> result ;
        solve( n , "" , result );

        return result ;
    }
};

File: 3211-generate-binary-strings-without-adjacent-zeros/3211-generate-binary-strings-without-adjacent-zeros.js
/**
 * @param {number} n
 * @return {string[]}
 */
var validStrings = function(n) {
    
    let result = []

    function solve( n , str , result ){

        if( str.length == n ){
            result.push(str);
            return
        }

        if( str.length == 0 || str[str.length-1] == '1' ){
            solve( n , str + '0' , result );
        }
        solve( n , str + '1' , result );

    }
    
    solve( n , "" , result );

    return result ;
};